zanata-war/src/main/java/org/zanata/rest/service/ResourceUtils.java

```diff
@@ -768,7 +773,7 @@ private void populateHeaderEntries(final List<HeaderEntry> headerEntries,
         }
 
         headerEntry = containedHeaders.get(PLURAL_FORMS_HDR);
-        if (headerEntry == null) {
+        if (headerEntry == null || headerEntry.getValue().isEmpty()) {
```

Comment:
I think a call to `trim()` here might be good too... unless the entry is being trimmed elsewhere already.